### ISSUE TITLE
Assembly name comparison

### ISSUE THREAD

**Issue by Sten-V** [toddams/RazorLight#439]
Is it possible that this call to `Contains(...)` instead should be a call to `Equals(...)`, or is this intentional?
https://github.com/toddams/RazorLight/blob/22381db30ca3c61b4b1e4004e1536407dd138e49/src/RazorLight/Compilation/DefaultMetadataReferenceManager.cs#L105

**Comment by toddams** [comment]
excludedAssemblies is a collection, assemblyRef.Name is a string. This is intentional

**Comment by Sten-V** [comment]
Yes, but the `Contains(...)` is not performed directly on that collection, it done on every string in that collection the way it looks now.
And because it is a `Contains(...)` comparison between two strings it's quite possible to exclude more assemblies than the consuming code intended.

**Comment by toddams** [comment]
Sorry, I actually overlooked the point. It's indeed weird, I'll double-check
